In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("stroke_risk_prediction_dataset.csv")
pd.set_option("display.max_columns",None)

In [3]:
df.head()

,Patient_ID,Age,Gender,Country,Height_cm,Weight_kg,BMI,Blood_Pressure_Systolic,Blood_Pressure_Diastolic,Heart_Rate,Blood_Glucose,HbA1c,Total_Cholesterol,HDL,LDL,Triglycerides,Smoking_Status,Alcohol_Consumption,Physical_Activity_Level,Sleep_Hours,Stress_Level,Diet_Quality,Family_History_Stroke,Family_History_Heart_Disease,Diabetes,Hypertension,Heart_Disease,Previous_TIA,Atrial_Fibrillation,Chronic_Kidney_Disease,Medication_Adherence,Exercise_Hours_Per_Week,Daily_Walking_Minutes,Work_Type,Residence_Type,Air_Pollution_Exposure,Stroke_Risk_Score,AI_Health_Recommendation,Doctor_Consultation_Needed,Stroke_Risk
0,1,80.0,Male,China,145.2,69.3,32.9,112,82,109,132.2,5.2,300.1,50.9,199.6,185.0,Former,Occasionally,Moderate,7.6,High,Healthy,No,No,No,No,No,No,No,No,Excellent,NaN,79.0,Retired,Urban,High,47,Exercise Regularly,Yes,Moderate
1,2,37.0,Female,United Kingdom,188.6,110.8,31.1,137,76,59,187.5,7.1,156.4,55.1,NaN,131.3,Current,Occasionally,High,7.3,Low,Poor,No,No,No,No,No,No,No,Yes,Average,0.6,48.0,Student,Rural,Medium,40,Improve Diet Quality,Yes,Moderate
2,3,85.0,Female,Germany,175.5,103.7,33.7,108,105,89,196.4,8.0,232.0,62.2,200.2,65.7,Former,Never,Low,6.3,High,Average,No,Yes,No,Yes,No,No,No,No,Poor,1.7,114.0,Private,Rural,Medium,76,Monitor Blood Glucose Daily,Yes,High
3,4,44.0,Other,Brazil,161.3,50.2,19.3,126,73,88,209.1,7.3,215.1,50.8,163.3,315.6,Never,Occasionally,High,9.9,Low,Healthy,No,No,No,No,No,No,No,Yes,Poor,1.5,101.0,Retired,Rural,Low,40,Exercise Regularly,Yes,Moderate
4,5,52.0,Female,United States,162.8,103.6,39.1,115,60,70,88.6,7.1,140.7,40.5,137.9,151.2,Never,Frequently,Moderate,9.4,High,Poor,Yes,No,No,No,No,No,No,No,Average,7.1,55.0,Government,Rural,Medium,38,Monitor Blood Pressure,Yes,Moderate


In [4]:
df["Stroke_Risk"].value_counts(dropna=False)

Stroke_Risk
Moderate    34375
High        12112
Low          3513
Name: count, dtype: int64

In [5]:
df.isna().sum()

Patient_ID                         0
Age                              219
Gender                             0
Country                            0
Height_cm                       1525
Weight_kg                       1523
BMI                                0
Blood_Pressure_Systolic            0
Blood_Pressure_Diastolic           0
Heart_Rate                         0
Blood_Glucose                    960
HbA1c                           1006
Total_Cholesterol               1004
HDL                             1021
LDL                             1050
Triglycerides                   1032
Smoking_Status                     0
Alcohol_Consumption                0
Physical_Activity_Level            0
Sleep_Hours                     1026
Stress_Level                       0
Diet_Quality                       0
Family_History_Stroke              0
Family_History_Heart_Disease       0
Diabetes                           0
Hypertension                       0
Heart_Disease                      0
P

In [6]:
pd.set_option("display.max_row",None)

In [16]:
#df["Age"].value_counts(dropna=False)

In [7]:
df["Age"] = df["Age"].fillna(round(df["Age"].mean(), 2))

In [8]:
df["Height_cm"] = df["Height_cm"].fillna(df["Height_cm"].mean())

In [9]:
#df["Height_cm"].head()

In [10]:
df["Weight_kg"] = df["Weight_kg"].fillna(df["Weight_kg"].mean())

In [11]:
#df["Weight_kg"].head()

In [12]:
df["Blood_Glucose"] = df["Blood_Glucose"].fillna(df["Blood_Glucose"].mean())

In [13]:
#df["Blood_Glucose"].head()

In [18]:
#df["HbA1c"].value_counts(dropna=False)

In [14]:
df["HbA1c"] = df["HbA1c"].fillna(df["HbA1c"].mean())

In [21]:
#df["Total_Cholesterol"].value_counts(dropna=False)

In [15]:
df["Total_Cholesterol"] = df["Total_Cholesterol"].fillna(df["Total_Cholesterol"].mean())

In [25]:
#df["HDL"].value_counts(dropna=False)

In [16]:
df["HDL"] = df["HDL"].fillna(df["HDL"].mean())

In [17]:
df["LDL"] = df["LDL"].fillna(df["LDL"].mean())

In [31]:
#df["Triglycerides"].value_counts(dropna=False)

In [18]:
df["Triglycerides"] = df["Triglycerides"].fillna(df["Triglycerides"].mean())

In [35]:
#df["Sleep_Hours"].value_counts(dropna=False)

In [19]:
df["Sleep_Hours"] = df["Sleep_Hours"].fillna(df["Sleep_Hours"].mean())

In [39]:
#df["Medication_Adherence"].value_counts(dropna=False)

In [20]:
df["Medication_Adherence"] = df["Medication_Adherence"].fillna(df["Medication_Adherence"].mode()[0])

In [47]:
#df["Exercise_Hours_Per_Week"].value_counts(dropna=False)

In [21]:
df["Exercise_Hours_Per_Week"] = df["Exercise_Hours_Per_Week"].fillna(df["Exercise_Hours_Per_Week"].mean())

In [51]:
#df["Daily_Walking_Minutes"].value_counts(dropna=False)

In [22]:
df["Daily_Walking_Minutes"] = df["Daily_Walking_Minutes"].fillna(df["Daily_Walking_Minutes"].mean())

In [23]:
df.isna().sum()

Patient_ID                      0
Age                             0
Gender                          0
Country                         0
Height_cm                       0
Weight_kg                       0
BMI                             0
Blood_Pressure_Systolic         0
Blood_Pressure_Diastolic        0
Heart_Rate                      0
Blood_Glucose                   0
HbA1c                           0
Total_Cholesterol               0
HDL                             0
LDL                             0
Triglycerides                   0
Smoking_Status                  0
Alcohol_Consumption             0
Physical_Activity_Level         0
Sleep_Hours                     0
Stress_Level                    0
Diet_Quality                    0
Family_History_Stroke           0
Family_History_Heart_Disease    0
Diabetes                        0
Hypertension                    0
Heart_Disease                   0
Previous_TIA                    0
Atrial_Fibrillation             0
Chronic_Kidney

In [24]:
df.head()

,Patient_ID,Age,Gender,Country,Height_cm,Weight_kg,BMI,Blood_Pressure_Systolic,Blood_Pressure_Diastolic,Heart_Rate,Blood_Glucose,HbA1c,Total_Cholesterol,HDL,LDL,Triglycerides,Smoking_Status,Alcohol_Consumption,Physical_Activity_Level,Sleep_Hours,Stress_Level,Diet_Quality,Family_History_Stroke,Family_History_Heart_Disease,Diabetes,Hypertension,Heart_Disease,Previous_TIA,Atrial_Fibrillation,Chronic_Kidney_Disease,Medication_Adherence,Exercise_Hours_Per_Week,Daily_Walking_Minutes,Work_Type,Residence_Type,Air_Pollution_Exposure,Stroke_Risk_Score,AI_Health_Recommendation,Doctor_Consultation_Needed,Stroke_Risk
0,1,80.0,Male,China,145.2,69.3,32.9,112,82,109,132.2,5.2,300.1,50.9,199.600000,185.0,Former,Occasionally,Moderate,7.6,High,Healthy,No,No,No,No,No,No,No,No,Excellent,5.001214,79.0,Retired,Urban,High,47,Exercise Regularly,Yes,Moderate
1,2,37.0,Female,United Kingdom,188.6,110.8,31.1,137,76,59,187.5,7.1,156.4,55.1,140.149685,131.3,Current,Occasionally,High,7.3,Low,Poor,No,No,No,No,No,No,No,Yes,Average,0.600000,48.0,Student,Rural,Medium,40,Improve Diet Quality,Yes,Moderate
2,3,85.0,Female,Germany,175.5,103.7,33.7,108,105,89,196.4,8.0,232.0,62.2,200.200000,65.7,Former,Never,Low,6.3,High,Average,No,Yes,No,Yes,No,No,No,No,Poor,1.700000,114.0,Private,Rural,Medium,76,Monitor Blood Glucose Daily,Yes,High
3,4,44.0,Other,Brazil,161.3,50.2,19.3,126,73,88,209.1,7.3,215.1,50.8,163.300000,315.6,Never,Occasionally,High,9.9,Low,Healthy,No,No,No,No,No,No,No,Yes,Poor,1.500000,101.0,Retired,Rural,Low,40,Exercise Regularly,Yes,Moderate
4,5,52.0,Female,United States,162.8,103.6,39.1,115,60,70,88.6,7.1,140.7,40.5,137.900000,151.2,Never,Frequently,Moderate,9.4,High,Poor,Yes,No,No,No,No,No,No,No,Average,7.100000,55.0,Government,Rural,Medium,38,Monitor Blood Pressure,Yes,Moderate


In [25]:
df.columns

Index(['Patient_ID', 'Age', 'Gender', 'Country', 'Height_cm', 'Weight_kg',
       'BMI', 'Blood_Pressure_Systolic', 'Blood_Pressure_Diastolic',
       'Heart_Rate', 'Blood_Glucose', 'HbA1c', 'Total_Cholesterol', 'HDL',
       'LDL', 'Triglycerides', 'Smoking_Status', 'Alcohol_Consumption',
       'Physical_Activity_Level', 'Sleep_Hours', 'Stress_Level',
       'Diet_Quality', 'Family_History_Stroke', 'Family_History_Heart_Disease',
       'Diabetes', 'Hypertension', 'Heart_Disease', 'Previous_TIA',
       'Atrial_Fibrillation', 'Chronic_Kidney_Disease', 'Medication_Adherence',
       'Exercise_Hours_Per_Week', 'Daily_Walking_Minutes', 'Work_Type',
       'Residence_Type', 'Air_Pollution_Exposure', 'Stroke_Risk_Score',
       'AI_Health_Recommendation', 'Doctor_Consultation_Needed',
       'Stroke_Risk'],
      dtype='object')

In [26]:
df.select_dtypes(include="object").columns

Index(['Gender', 'Country', 'Smoking_Status', 'Alcohol_Consumption',
       'Physical_Activity_Level', 'Stress_Level', 'Diet_Quality',
       'Family_History_Stroke', 'Family_History_Heart_Disease', 'Diabetes',
       'Hypertension', 'Heart_Disease', 'Previous_TIA', 'Atrial_Fibrillation',
       'Chronic_Kidney_Disease', 'Medication_Adherence', 'Work_Type',
       'Residence_Type', 'Air_Pollution_Exposure', 'AI_Health_Recommendation',
       'Doctor_Consultation_Needed', 'Stroke_Risk'],
      dtype='object')

In [34]:
x = df.drop([
    "Patient_ID",
    "Stroke_Risk",
    "Stroke_Risk_Score",
    "AI_Health_Recommendation",
    "Doctor_Consultation_Needed"
], axis=1)

y = df["Stroke_Risk"]

In [35]:
categorical_cols = [
    'Gender', 
    'Country', 
    'Smoking_Status', 
    'Alcohol_Consumption',
    'Physical_Activity_Level', 
    'Stress_Level',
    'Diet_Quality',
    'Family_History_Stroke',
    'Family_History_Heart_Disease',
    'Diabetes',
    'Hypertension',
    'Heart_Disease', 
    'Previous_TIA',
    'Atrial_Fibrillation',
    'Chronic_Kidney_Disease',
    'Medication_Adherence', 
    'Work_Type',
    'Residence_Type', 
    'Air_Pollution_Exposure'
]

In [36]:
x = pd.get_dummies(x,columns=categorical_cols,drop_first=True,dtype=int)

In [37]:
x.head()

,Age,Height_cm,Weight_kg,BMI,Blood_Pressure_Systolic,Blood_Pressure_Diastolic,Heart_Rate,Blood_Glucose,HbA1c,Total_Cholesterol,HDL,LDL,Triglycerides,Sleep_Hours,Exercise_Hours_Per_Week,Daily_Walking_Minutes,Gender_Male,Gender_Other,Country_Bangladesh,Country_Brazil,Country_Canada,Country_China,Country_France,Country_Germany,Country_India,Country_Indonesia,Country_Italy,Country_Japan,Country_Malaysia,Country_Mexico,Country_Nigeria,Country_Pakistan,Country_Russia,Country_Saudi Arabia,Country_Singapore,Country_South Africa,Country_South Korea,Country_Spain,Country_Turkey,Country_UAE,Country_United Kingdom,Country_United States,Smoking_Status_Former,Smoking_Status_Never,Alcohol_Consumption_Never,Alcohol_Consumption_Occasionally,Physical_Activity_Level_Low,Physical_Activity_Level_Moderate,Stress_Level_Low,Stress_Level_Moderate,Diet_Quality_Healthy,Diet_Quality_Poor,Family_History_Stroke_Yes,Family_History_Heart_Disease_Yes,Diabetes_Yes,Hypertension_Yes,Heart_Disease_Yes,Previous_TIA_Yes,Atrial_Fibrillation_Yes,Chronic_Kidney_Disease_Yes,Medication_Adherence_Excellent,Medication_Adherence_Poor,Work_Type_Private,Work_Type_Retired,Work_Type_Self-Employed,Work_Type_Student,Residence_Type_Urban,Air_Pollution_Exposure_Low,Air_Pollution_Exposure_Medium
0,80.0,145.2,69.3,32.9,112,82,109,132.2,5.2,300.1,50.9,199.600000,185.0,7.6,5.001214,79.0,1,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,1,0,1,0,0,1,0,0,0,0,0,0,0,0,0,1,0,0,1,0,0,1,0,0
1,37.0,188.6,110.8,31.1,137,76,59,187.5,7.1,156.4,55.1,140.149685,131.3,7.3,0.600000,48.0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,1,0,0,1,0,0,1,0,0,0,0,0,0,0,1,0,0,0,0,0,1,0,0,1
2,85.0,175.5,103.7,33.7,108,105,89,196.4,8.0,232.0,62.2,200.200000,65.7,6.3,1.700000,114.0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,1,0,1,0,0,0,0,0,0,1,0,1,0,0,0,0,0,1,1,0,0,0,0,0,1
3,44.0,161.3,50.2,19.3,126,73,88,209.1,7.3,215.1,50.8,163.300000,315.6,9.9,1.500000,101.0,0,1,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,1,0,0,1,0,1,0,0,0,0,0,0,0,0,1,0,1,0,1,0,0,0,1,0
4,52.0,162.8,103.6,39.1,115,60,70,88.6,7.1,140.7,40.5,137.900000,151.2,9.4,7.100000,55.0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,1,0,0,0,1,0,0,0,1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1


In [38]:
from sklearn.model_selection import train_test_split

In [39]:
x_train, x_test, y_train, y_test = train_test_split(
    x,y,test_size = 0.30, stratify = y, random_state = 42
)

In [40]:
x_train.head()

,Age,Height_cm,Weight_kg,BMI,Blood_Pressure_Systolic,Blood_Pressure_Diastolic,Heart_Rate,Blood_Glucose,HbA1c,Total_Cholesterol,HDL,LDL,Triglycerides,Sleep_Hours,Exercise_Hours_Per_Week,Daily_Walking_Minutes,Gender_Male,Gender_Other,Country_Bangladesh,Country_Brazil,Country_Canada,Country_China,Country_France,Country_Germany,Country_India,Country_Indonesia,Country_Italy,Country_Japan,Country_Malaysia,Country_Mexico,Country_Nigeria,Country_Pakistan,Country_Russia,Country_Saudi Arabia,Country_Singapore,Country_South Africa,Country_South Korea,Country_Spain,Country_Turkey,Country_UAE,Country_United Kingdom,Country_United States,Smoking_Status_Former,Smoking_Status_Never,Alcohol_Consumption_Never,Alcohol_Consumption_Occasionally,Physical_Activity_Level_Low,Physical_Activity_Level_Moderate,Stress_Level_Low,Stress_Level_Moderate,Diet_Quality_Healthy,Diet_Quality_Poor,Family_History_Stroke_Yes,Family_History_Heart_Disease_Yes,Diabetes_Yes,Hypertension_Yes,Heart_Disease_Yes,Previous_TIA_Yes,Atrial_Fibrillation_Yes,Chronic_Kidney_Disease_Yes,Medication_Adherence_Excellent,Medication_Adherence_Poor,Work_Type_Private,Work_Type_Retired,Work_Type_Self-Employed,Work_Type_Student,Residence_Type_Urban,Air_Pollution_Exposure_Low,Air_Pollution_Exposure_Medium
24890,35.0,151.4,103.9,45.3,154,78,79,215.0,9.000000,203.8,75.7,140.2,285.900000,5.4,1.3,3.0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,1,0,1,0,1,0,1,0,0,0,0,0,0,0,0,0,1,0,0,1,0,0,0,1
17796,85.0,173.9,54.5,18.0,135,103,94,184.8,7.400000,149.0,43.6,183.8,263.400000,5.8,9.7,49.0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,1,1,0,1,0,0,0,1,0,0,0,0,0,1,0,1,0,0,0,0,0,0
17565,66.0,178.5,112.9,35.4,163,107,108,171.3,8.200000,312.0,47.4,118.1,229.700000,6.2,8.2,94.0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,1,0,1,0,0,1,0,1,0,1,1,0,0,1,0,0,0,0,1,0,0,1,0,0,1,0,1
6171,61.0,190.4,63.9,17.6,154,65,71,138.5,7.488103,231.7,41.1,178.7,297.700000,6.5,7.0,47.0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,1,0,0,0,0,1,0,0,0,0,0,1,0,0,1,0,1,0,1
11591,56.0,152.4,69.8,30.1,158,74,85,87.9,8.000000,308.8,29.7,176.4,205.200147,6.5,1.6,82.0,1,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,1,0,0,0,0,0,0,0,0,0,0,1,0,1,0,0,0,0,1


In [41]:
x_test.head()

,Age,Height_cm,Weight_kg,BMI,Blood_Pressure_Systolic,Blood_Pressure_Diastolic,Heart_Rate,Blood_Glucose,HbA1c,Total_Cholesterol,HDL,LDL,Triglycerides,Sleep_Hours,Exercise_Hours_Per_Week,Daily_Walking_Minutes,Gender_Male,Gender_Other,Country_Bangladesh,Country_Brazil,Country_Canada,Country_China,Country_France,Country_Germany,Country_India,Country_Indonesia,Country_Italy,Country_Japan,Country_Malaysia,Country_Mexico,Country_Nigeria,Country_Pakistan,Country_Russia,Country_Saudi Arabia,Country_Singapore,Country_South Africa,Country_South Korea,Country_Spain,Country_Turkey,Country_UAE,Country_United Kingdom,Country_United States,Smoking_Status_Former,Smoking_Status_Never,Alcohol_Consumption_Never,Alcohol_Consumption_Occasionally,Physical_Activity_Level_Low,Physical_Activity_Level_Moderate,Stress_Level_Low,Stress_Level_Moderate,Diet_Quality_Healthy,Diet_Quality_Poor,Family_History_Stroke_Yes,Family_History_Heart_Disease_Yes,Diabetes_Yes,Hypertension_Yes,Heart_Disease_Yes,Previous_TIA_Yes,Atrial_Fibrillation_Yes,Chronic_Kidney_Disease_Yes,Medication_Adherence_Excellent,Medication_Adherence_Poor,Work_Type_Private,Work_Type_Retired,Work_Type_Self-Employed,Work_Type_Student,Residence_Type_Urban,Air_Pollution_Exposure_Low,Air_Pollution_Exposure_Medium
33288,67.0,154.700000,79.5,33.2,117,108,61,152.4,9.9,244.9,51.4,129.4,132.5,5.4,3.4,34.0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,1,0,0,1,0,0,1,0,0,1,0,0,0,0,1,0,0,0,0,0,0,0,0,1,0
26674,50.0,170.125582,105.0,35.2,135,66,93,187.5,4.8,228.5,47.4,153.0,201.4,9.2,9.2,83.0,0,1,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,1,0,1,0,1,0,1,0,0,0,1,0,0,0,0,0,0,0,1,0,0,1,1,0
11036,53.9,192.200000,63.8,17.3,112,104,64,136.4,7.0,142.1,61.0,168.2,77.7,4.4,6.1,86.0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,1,1,0,1,0,0,0,1,0,0,0,0,0,1,0,0,0,0,1,1,0,1
5500,67.0,148.200000,83.8,38.2,110,92,80,112.3,7.0,210.0,37.5,102.9,161.6,8.5,10.0,7.0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,1,1,0,0,1,0,0,0,0,0,0,0,0,0,0,0,1,0,0,1,1,0
26586,50.0,179.500000,75.4,23.4,165,65,101,101.9,6.3,308.1,43.2,193.3,85.0,7.8,4.5,59.0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,1,0,0,0,1,1,0,0,0,1,0,0,1,0,0,0,0,0,1,0,0,1,0,0,0,0


In [42]:
y_test.head()

33288    Moderate
26674    Moderate
11036    Moderate
5500     Moderate
26586    Moderate
Name: Stroke_Risk, dtype: object

In [43]:
from sklearn.preprocessing import StandardScaler

In [44]:
scaler = StandardScaler()

In [45]:
x_train_scaled = scaler.fit_transform(x_train)
x_test_scaled = scaler.transform(x_test)

In [46]:
from sklearn.multiclass import OneVsRestClassifier

In [47]:
from sklearn.linear_model import LogisticRegression

In [59]:
model = OneVsRestClassifier(
     LogisticRegression(
         max_iter = 1000,
         class_weight="balanced")
)

In [60]:
model

OneVsRestClassifier(estimator=LogisticRegression(class_weight='balanced',
                                                 max_iter=1000))

In [61]:
model.fit(x_train_scaled,y_train)

C:\ProgramData\anaconda3\Lib\site-packages\sklearn\linear_model\_logistic.py:451: OptimizeWarning: Unknown solver options: iprint
  opt_res = optimize.minimize(
C:\ProgramData\anaconda3\Lib\site-packages\sklearn\linear_model\_logistic.py:451: OptimizeWarning: Unknown solver options: iprint
  opt_res = optimize.minimize(
C:\ProgramData\anaconda3\Lib\site-packages\sklearn\linear_model\_logistic.py:451: OptimizeWarning: Unknown solver options: iprint
  opt_res = optimize.minimize(


OneVsRestClassifier(estimator=LogisticRegression(class_weight='balanced',
                                                 max_iter=1000))

In [62]:
y_pred = model.predict(x_test_scaled)

In [63]:
y_pred

array(['High', 'Moderate', 'Low', ..., 'High', 'Moderate', 'Moderate'],
      shape=(15000,), dtype='<U8')

In [64]:
comparison = pd.DataFrame(
    {
        "Actual": y_test,
        "Predicted": y_pred
    }
)
comparison.head(15)

,Actual,Predicted
33288,Moderate,High
26674,Moderate,Moderate
11036,Moderate,Low
5500,Moderate,Moderate
26586,Moderate,Moderate
2916,Moderate,Moderate
22193,Moderate,Moderate
24130,Moderate,Moderate
25277,Moderate,High
27382,Moderate,Moderate


In [65]:
from sklearn.metrics import accuracy_score, confusion_matrix, classification_report

In [66]:
print("Accuracy :", accuracy_score(y_test,y_pred)*100)

Accuracy : 80.63333333333334


In [67]:
cm = confusion_matrix(y_test,y_pred)

In [69]:
cm

array([[3318,    0,  316],
       [   0,  874,  180],
       [1595,  814, 7903]])

In [70]:
print("classification Report :\n", classification_report(y_test,y_pred))

classification Report :
               precision    recall  f1-score   support

        High       0.68      0.91      0.78      3634
         Low       0.52      0.83      0.64      1054
    Moderate       0.94      0.77      0.84     10312

    accuracy                           0.81     15000
   macro avg       0.71      0.84      0.75     15000
weighted avg       0.85      0.81      0.81     15000



**Conclusion:** The balanced Logistic Regression model achieved 81% accuracy with a 75% Macro F1-score. The model achieved 91% recall for High-risk and 83% recall for Low-risk, demonstrating improved identification of different stroke-risk categories despite class imbalance.

**Final metrics:**

- Accuracy: 81%
- Macro Precision: 71%
- Macro Recall: 84%
- Macro F1-Score: 75%
- Weighted F1-Score: 81%